tests/testthat/test-kobo.R

```diff
@@ -100,3 +100,82 @@ test_that("non-existing kobo host throws error", {
   kobo <- Kobo$new(base_url_v2 = "https://nokobohere.correlaid.org", kobo_token = Sys.getenv("KBTBR_TOKEN"))
   expect_error(kobo$get("assets/"), regexp = "^SSL.+certificate.+")
 })
+
+
+#' -----------------------------------------------------------------------------
+#' Testing POST methods
+test_that("Kobo$post can clone assets", {
+    vcr::use_cassette("kobo-clone-assets-simple-post", {
+        kobo <- Kobo$new(base_url_v2 = BASE_URL, kobo_token = Sys.getenv("KBTBR_TOKEN"))
+        clone_asset <- kobo$post("assets/", body = list("clone_from" = "a84jmwwdPEsZMhK7s2i4SL",
+                                                        "name" = "vcr_test_name",
+                                                        "asset_type" = "survey"))
+    })
+    expect_equal(clone_asset$url,"https://kobo.correlaid.org/api/v2/assets/")
+    expect_equal(clone_asset$method,"post")
+    expect_equal(clone_asset$status_code,201)
+    expect_equal(clone_asset$success(),TRUE)
+    expect_equal(clone_asset$status_http()$message,"Created")
+    expect_equal(clone_asset$status_http()$explanation,"Document created, URL follows")
+})
+
+test_that("kobo$clone_asset can clone assets", {
+    vcr::use_cassette("kobo-post-clone-asset", {
+        kobo <- suppressWarnings(Kobo$new(base_url_v2 = BASE_URL, kobo_token = Sys.getenv("KBTBR_TOKEN")))
+        clone_asset <- kobo$clone_asset(clone_from = "a84jmwwdPEsZMhK7s2i4SL",
+                                        name = "vcr_test_name",
+                                        asset_type = "survey")
+        })
+    expect_equal(clone_asset$url,"https://kobo.correlaid.org/api/v2/assets/")
+    expect_equal(clone_asset$method,"post")
+    expect_equal(clone_asset$status_code,201)
+    expect_equal(clone_asset$success(),TRUE)
+    expect_equal(clone_asset$status_http()$message,"Created")
+    expect_equal(clone_asset$status_http()$explanation,"Document created, URL follows")
+})
+
+test_that("kobo$deploy_asset can deploy assets", {
+    vcr::use_cassette("kobo-post-deploy-asset", {
+        kobo <- suppressWarnings(Kobo$new(base_url_v2 = BASE_URL, kobo_token = Sys.getenv("KBTBR_TOKEN")))
+        deploy_asset <- kobo$deploy_asset(uid = "aQVGH8G68EP737tDBABRwC")
+    })
+    expect_equal(deploy_asset$url,
+                 "https://kobo.correlaid.org/api/v2/assets/aQVGH8G68EP737tDBABRwC/deployment/")
+    expect_equal(deploy_asset$method,"post")
+    expect_equal(deploy_asset$status_code,200) # i don't understand why not 201. but with 200 it successfully deployed
```

Comment:
Hah super weird! :mag: 
Maybe this is something we could bring to the kpi github repo? :thinking:

Reply:
maybe